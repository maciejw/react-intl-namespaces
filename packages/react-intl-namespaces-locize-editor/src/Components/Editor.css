body.searchEnabled * {
  pointer-events: auto;
}
