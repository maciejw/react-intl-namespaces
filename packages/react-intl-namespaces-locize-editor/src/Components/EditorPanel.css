.button {
  border: none;
  color: #fff;
  cursor: pointer;
  font-family: Helvetica, Arial, sans-serif;
  font-size: 14px;
  font-weight: 300;
  height: 30px;
  line-height: 30px;
  min-width: 90px;
  outline: none;
  padding: 0 2px;
  text-align: center;
  text-decoration: none;
  text-overflow: ellipsis;
  text-transform: uppercase;
  white-space: nowrap;
}
.button:hover {
  opacity: 0.9;
}
.button:active {
  padding: 0 1px 0 3px;
}
.buttonOff {
  background-color: #d50000;
}
.buttonOn {
  background-color: #54a229;
}
.option {
  text-transform: uppercase;
}
.panel {
  background-color: #fff;
  border: solid 1px #1976d2;
  bottom: 2px;
  box-shadow: 0px 1px 2px 0px rgba(0, 0, 0, 0.5);
  font-family: Helvetica, Arial, sans-serif;
  padding: 10px;
  position: fixed;
  right: 18px;
  z-index: 2001;
  user-select: none;
}
.panel.pinned {
  max-width: 1000px;
}
.panel .content {
  display: none;
  float: left;
}
.panel.pinned .content {
  display: block;
  width: auto;
}
.select {
  width: 50px;
  border: none;
  box-sizing: content-box;
  font-family: Helvetica, Arial, sans-serif;
  font-size: 14px;
  font-weight: 300;
  padding: 6px;
  text-transform: uppercase;
}

.title {
  color: #1976d2;
  font-family: Helvetica, Arial, sans-serif;
  font-size: 14px;
  font-weight: 300;
  margin: 0 0 5px 0;
}

.pin {
  height: 52px;
  min-width: 20px;
  float: right;
  margin-left: 5px;
}
.pin:active {
  padding: initial;
}
.pin::after {
  font-size: 30px;
}
.pin.pinned::after {
  content: '\203A';
}
.pin.unpinned::after {
  content: '\2039';
}
