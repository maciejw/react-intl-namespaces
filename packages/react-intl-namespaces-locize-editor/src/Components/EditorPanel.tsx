import classNames from 'classnames';
import * as React from 'react';
import * as ReactDOM from 'react-dom';
import styles from './EditorPanel.css';

export class EditorPanel extends React.Component<
  EditorPanel.Props,
  EditorPanel.State
> {
  constructor(props: EditorPanel.Props, state: EditorPanel.State) {
    super(props, state);
  }

  public render() {
    const languages = this.props.languages;
    return (
      <div
        className={classNames(styles.panel, {
          [styles.pinned]: this.props.pinned,
        })}
      >
        {
          <span className={classNames(styles.content)}>
            <div className={styles.title}>locize editor</div>
            {languages.length > 0 && (
              <select
                className={styles.select}
                value={this.props.language}
                onChange={e => this.props.onChangeLanguage(e.target.value)}
              >
                {languages.map(l => (
                  <option key={l} value={l} className={styles.option}>
                    {l}
                  </option>
                ))}
              </select>
            )}{' '}
            <button
              onClick={() => this.props.onToggleSearch()}
              className={classNames(styles.button, {
                [styles.buttonOn]: this.props.searchEnabled,
                [styles.buttonOff]: !this.props.searchEnabled,
              })}
            >
              {this.props.searchEnabled ? 'On' : 'Off'}
            </button>{' '}
            <button
              onClick={() => this.props.onShowIds()}
              className={classNames(styles.button, {
                [styles.buttonOn]: this.props.showIds,
                [styles.buttonOff]: !this.props.showIds,
              })}
            >
              id's
            </button>{' '}
            <button
              onClick={() => this.props.onRefresh()}
              className={classNames(styles.button, styles.buttonOn)}
            >
              refresh
            </button>{' '}
          </span>
        }

        <button
          className={classNames(styles.button, styles.pin, {
            [styles.pinned]: this.props.pinned,
            [styles.buttonOn]: this.props.pinned,
            [styles.unpinned]: !this.props.pinned,
            [styles.buttonOff]: !this.props.pinned,
          })}
          onClick={() => this.props.onTogglePinned()}
        >
          {' '}
        </button>
      </div>
    );
  }
}
export namespace EditorPanel {
  export interface State {}
  export interface Props {
    searchEnabled: boolean;
    showIds: boolean;
    onTogglePinned: () => void;
    pinned: boolean;
    onRefresh: () => void;
    onToggleSearch: () => void;
    onShowIds: () => void;
    language: string;
    onChangeLanguage: (language: string) => void;
    languages: string[];
  }
}
