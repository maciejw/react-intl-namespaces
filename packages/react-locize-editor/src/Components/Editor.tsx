import * as React from 'react';
import { LocizeEditorBinding } from '../locizeEditorBinding';

const editor = new LocizeEditorBinding({
  backend: {
    apiKey: '6b9ac145-b395-47dc-bcc4-001398c4c843',
    projectId: '2592abb8-1129-457d-a06b-836745d33c55',
    referenceLng: 'en',
  },
  enabled: true,
});

export class Editor extends React.Component {
  constructor(props: {}, context: {}) {
    super(props, context);
  }
  public render() {
    return null;
  }
}
