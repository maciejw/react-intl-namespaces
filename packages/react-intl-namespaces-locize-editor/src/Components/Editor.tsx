import * as React from 'react';
import * as ReactDOM from 'react-dom';

import { logger } from 'react-intl-namespaces';
import { DOMHelpers } from '../DOMHelpers';
import { EditorPanel } from './EditorPanel';
import { EditorWindow } from './EditorWindow';

export class Editor extends React.Component<
  Editor.RequiredProps & Partial<Editor.OptionalProps>
> {
  public render() {
    return <EditorComponent {...Editor.defaultProps} {...this.props} />;
  }
}
export class EditorComponent extends React.Component<
  Editor.RequiredProps & Editor.OptionalProps,
  Editor.State
> {
  private message: (ev: MessageEvent) => void;
  // prettier-ignore
  private keypress: (ev: KeyboardEvent) => void;
  // prettier-ignore
  private click: (e: MouseEvent) => void;
  // prettier-ignore
  private postMessage!: Promise<EditorWindow.PostMessage>;
  // prettier-ignore

  private editor: HTMLElement;

  constructor(props: Editor.RequiredProps & Editor.OptionalProps, context: {}) {
    super(props, context);
    this.state = {
      searchEnabled: true,
      showEditor: this.props.enabled,
      showIds: false,
    };

    this.click = (ev: MouseEvent) => {
      this.lookupInEditor(ev);
    };

    const { toggleKeyModifier, toggleKey } = this.props;

    this.keypress = (ev: KeyboardEvent) => {
      if (ev[toggleKeyModifier] && ev.key === toggleKey) {
        this.onSearchEnabled();
      }
    };
    this.message = (ev: MessageEvent) => {
      if (ev.data[toggleKeyModifier] && ev.data.key === toggleKey) {
        this.onSearchEnabled();
      }
    };

    this.editor = DOMHelpers.Editor.createTargetElement(
      this.props.document,
      this.props.editorId,
    );

    props.document.body.addEventListener('click', this.click);
    props.document.addEventListener('keypress', this.keypress);
    props.window.addEventListener('message', this.message);
  }

  public componentWillUnmount() {
    this.props.document.body.removeChild(this.editor);
    this.props.document.body.removeEventListener('click', this.click);
    this.props.document.removeEventListener('keypress', this.keypress);
    this.props.window.removeEventListener('message', this.message);
  }

  public render() {
    const { searchEnabled, showIds } = this.state;
    return ReactDOM.createPortal(
      <div data-ignore-editor="true">
        <EditorPanel
          showIds={showIds}
          searchEnabled={searchEnabled}
          onRefresh={() => this.onRefresh()}
          onSearchEnabled={() => this.onSearchEnabled()}
          onShowIds={() => this.onShowIds()}
          language={this.props.language}
          onChangeLanguage={this.props.onChangeLanguage}
          getLanguages={this.props.getLanguages}
        />
        <EditorWindow
          mode={this.props.mode}
          editorWidthInPixels={this.props.editorWidthInPixels}
          windowOpenTimeout={3000}
          window={window}
          url={this.props.url}
          onOpen={callback => this.open(callback)}
        />
      </div>,
      this.editor,
    );
  }

  private onShowIds() {
    const { showIds: oldShowIds, ...rest } = this.state;
    const showIds = !oldShowIds;
    this.setState({ showIds, ...rest });
    this.props.onShowIds(showIds);
  }
  private open(callback: Promise<EditorWindow.PostMessage>) {
    this.postMessage = callback;
  }

  private async lookupInEditor(e: MouseEvent) {
    if (!this.state.searchEnabled) {
      return;
    }
    e.preventDefault();

    const resourceContainer = DOMHelpers.Editor.locateResourceContext(e.target);

    if (resourceContainer) {
      const { namespace, key, defaultMessage, description } = resourceContainer;
      const message = this.createMessage(
        namespace,
        key,
        defaultMessage,
        description,
      );
      const postMessage = await this.postMessage;

      postMessage(message, this.props.url);
    } else {
      logger.warn(
        `Missing key and namespace of resource, try search for a text or select ID mode`,
      );
    }
  }
  private createMessage(
    ns: string,
    key: string,
    defaultMessage?: string,
    description?: string,
  ): Editor.SearchMessage {
    const { language: lng, projectId, version } = this.props;
    return {
      lng,
      message: 'searchForKey',
      ns,
      projectId,
      token: key,
      version,
    };
  }

  private onSearchEnabled() {
    const { searchEnabled } = this.state;
    this.setState({ searchEnabled: !searchEnabled });
  }
  private onRefresh() {
    this.props.onRefresh();
  }
}

export namespace Editor {
  export interface State {
    searchEnabled: boolean;
    showIds: boolean;
    showEditor: boolean;
  }
  export const defaultProps: Editor.OptionalProps = {
    document,
    editorId: 'locize-editor',
    editorWidthInPixels: 700,
    enabled: false,
    getLanguages: () => [],
    mode: 'iframe',
    onChangeLanguage: language => void 0,
    onRefresh: () => void 0,
    onShowIds: show => void 0,
    toggleKey: 'Enter',
    toggleKeyModifier: 'ctrlKey',
    url: 'https://www.locize.io',
    version: 'latest',
    window,
  };

  export interface OptionalProps {
    window: Window;
    document: Document;
    editorId: string;
    editorWidthInPixels: number;
    enabled: boolean;
    toggleKey: string;
    toggleKeyModifier: 'ctrlKey' | 'altKey' | 'shiftKey';
    mode: 'window' | 'iframe';
    url: string;
    version: string;
    onShowIds: (show: boolean) => void;
    onChangeLanguage: (language: string) => void;
    getLanguages: () => string[];
    onRefresh: () => void;
  }

  export interface RequiredProps {
    apiKey: string;
    projectId: string;
    referenceLanguage: string;
    language: string;
  }
  export interface SearchMessage {
    message: 'searchForKey';
    projectId: string;
    version: string;
    lng: string;
    ns: string;
    token: string;
    defaultMessage?: string;
    description?: string;
  }
}
