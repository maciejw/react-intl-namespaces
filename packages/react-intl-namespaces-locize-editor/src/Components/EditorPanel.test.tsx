import { mount } from 'enzyme';
import * as React from 'react';
import { Editor } from './Editor';
import { EditorPanel } from './EditorPanel';

const options: Editor.RequiredProps = {
  apiKey: '6b9ac145-b395-47dc-bcc4-001398c4c843',
  language: 'en',
  projectId: '2592abb8-1129-457d-a06b-836745d33c55',
  referenceLanguage: 'en',
};
describe('EditorPanel', () => {
  it('should show panel and handle events', () => {
    const getLanguagesMock = jest.fn();

    const PropsMock = jest.fn<EditorPanel.Props>(() => ({
      getLanguages: jest.fn().mockReturnValue(['en', 'pl']),
      onChangeLanguage: jest.fn(),
      onRefresh: jest.fn(),
      onSearchEnabled: jest.fn(),
      onShowIds: jest.fn(),
    }));

    const mock = new PropsMock();

    const wrapper = mount(
      <EditorPanel
        showIds={true}
        searchEnabled={true}
        language="en"
        {...mock}
      />,
    );
    expect(wrapper.html()).toMatchSnapshot();

    wrapper.find('button').forEach(b => b.simulate('click'));
    wrapper.find('select').forEach(b => b.simulate('change'));

    expect(wrapper.html()).toMatchSnapshot();

    expect(mock).toMatchSnapshot();
  });
});
