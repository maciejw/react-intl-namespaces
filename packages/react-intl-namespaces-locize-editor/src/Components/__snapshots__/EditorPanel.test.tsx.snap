// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditorPanel should not show editor 1`] = `"<div class=\\"panel pinned\\"><span class=\\"content\\"><div class=\\"title\\">locize editor</div><select class=\\"select\\"><option value=\\"en\\" class=\\"option\\">en</option><option value=\\"pl\\" class=\\"option\\">pl</option></select> <button class=\\"button buttonOff\\">Off</button> <button class=\\"button buttonOn\\">id's</button> <button class=\\"button buttonOn\\">refresh</button> </span><button class=\\"button pin pinned buttonOn\\"> </button></div>"`;

exports[`EditorPanel should show panel and handle events 1`] = `"<div class=\\"panel pinned\\"><span class=\\"content\\"><div class=\\"title\\">locize editor</div><select class=\\"select\\"><option value=\\"en\\" class=\\"option\\">en</option><option value=\\"pl\\" class=\\"option\\">pl</option></select> <button class=\\"button buttonOn\\">On</button> <button class=\\"button buttonOn\\">id's</button> <button class=\\"button buttonOn\\">refresh</button> </span><button class=\\"button pin pinned buttonOn\\"> </button></div>"`;

exports[`EditorPanel should show panel and handle events 2`] = `"<div class=\\"panel\\"><span class=\\"content\\"><div class=\\"title\\">locize editor</div><select class=\\"select\\"><option value=\\"en\\" class=\\"option\\">en</option><option value=\\"pl\\" class=\\"option\\">pl</option></select> <button class=\\"button buttonOn\\">On</button> <button class=\\"button buttonOn\\">id's</button> <button class=\\"button buttonOn\\">refresh</button> </span><button class=\\"button pin unpinned buttonOff\\"> </button></div>"`;

exports[`EditorPanel should show panel and handle events 3`] = `
Object {
  "getLanguages": [MockFunction] {
    "calls": Array [
      Array [],
      Array [],
    ],
  },
  "onChangeLanguage": [MockFunction] {
    "calls": Array [
      Array [
        "en",
      ],
    ],
  },
  "onRefresh": [MockFunction] {
    "calls": Array [
      Array [],
    ],
  },
  "onSearchEnabled": [MockFunction] {
    "calls": Array [
      Array [],
    ],
  },
  "onShowIds": [MockFunction] {
    "calls": Array [
      Array [],
    ],
  },
}
`;
